x/shared/keeper/msg_server_update_param_test.go

```diff
@@ -103,8 +118,15 @@ func TestMsgUpdateParam_UpdateProofWindowOpenOffsetBlocks(t *testing.T) {
 	k, ctx := testkeeper.SharedKeeper(t)
 	msgSrv := keeper.NewMsgServerImpl(k)
 
-	// Set the parameters to their default values
 	defaultParams := sharedtypes.DefaultParams()
+
+	// Get the difference between the previous proof window open offset blocks and its
+	// new value to update the supplier unbonding period blocks such as it is greater
+	// than the cumulative proof window close blocks to pass UpdateParam validation.
+	paramDelta := uint64(expectedProofWindowOpenOffestBlocks) - defaultParams.ProofWindowOpenOffsetBlocks
+	defaultParams.SupplierUnbondingPeriodSessions = getMinSupplierUnbondingPeriodSessions(&defaultParams, paramDelta)
```

Comment:
NIT: What do you think of capturing the `paramDelta` calculation inside of `getMinSupplier...`.

It already takes in `defaultParams` so you'd just provide `expectedProofWindow...` instead?

Reply:
In order to achieve that, we need to know which variable is affected.

We currently do: `fn([A, B, C], X-B)`
Which we can't replicate by doing `fn([A, B, C], X)` since we lose which item we are subtracting from.
We can achieve by passing the affected original value `B` though. `fn([A, B, C], X, B)`